lntest/mock/walletcontroller.go

```diff
@@ -63,9 +63,19 @@ func (w *WalletController) ConfirmedBalance(confs int32,
 func (w *WalletController) NewAddress(addrType lnwallet.AddressType,
 	change bool, _ string) (btcutil.Address, error) {
 
-	addr, _ := btcutil.NewAddressPubKey(
-		w.RootKey.PubKey().SerializeCompressed(), &chaincfg.MainNetParams,
-	)
+	key := w.RootKey.PubKey().SerializeCompressed()
+
+	if addrType == lnwallet.WitnessPubKey {
+		witProgram := btcutil.Hash160(key)
+
+		addr, _ := btcutil.NewAddressWitnessPubKeyHash(
```

Comment:
why swallow the `error` if we might as well just return it?

Reply:
returning error